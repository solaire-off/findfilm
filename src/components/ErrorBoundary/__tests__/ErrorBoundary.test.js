import React from "react";
import { ErrorBoundary } from "..";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";

const Something = () => <span>Some working code here</span>;

it("Current render without error", () => {
  const { asFragment } = render(
    <ErrorBoundary>
      <Something />
    </ErrorBoundary>
  );

  expect(asFragment()).toMatchSnapshot();
});

it("Should display an error message if component throws", () => {
  const error = new Error("Test");

  const wrapper = shallow(
    <ErrorBoundary>
      <Something />
    </ErrorBoundary>
  );
  wrapper.find(Something).simulateError(error);

  expect(wrapper.text()).toContain("Oops");
});
