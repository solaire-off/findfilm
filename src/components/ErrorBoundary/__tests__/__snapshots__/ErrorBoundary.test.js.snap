// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render without error 1`] = `
<DocumentFragment>
  <span>
    Some working code here
  </span>
</DocumentFragment>
`;
